### PR TITLE
Use generic completeness/soundness tests

### PR DESCRIPTION
**Pull request by volhovm**
Reduces test code duplication by using generic completeness and soundness tests.

### CODE REVIEW

**Comment by volhovm** [comment]
The CI fails with a weird issue:

`Canceling due to signal: 4 tests still running`

Did the test time configuration change? :thinking:

**Comment by volhovm** [comment]
OK wait maybe it was a CI bug. I restarted it again now and /now/ it seems to pass?...

**Review by querolita — APPROVED** [review]

File: msm/src/test/generic.rs [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    expr::E, logup::LookupTableID, lookups::LookupTableIDs, proof::ProofInputs, prover::prove,
+    verifier::verify, witness::Witness, BaseSponge, Fp, OpeningProof, ScalarSponge, BN254,
+};
+use ark_ff::Zero;
+use kimchi::circuits::domains::EvaluationDomains;
+use poly_commitment::pairing_proof::PairingSRS;
+use rand::{CryptoRng, RngCore};
+
+/// No lookups, no selectors, only witness column. `N_WIT == N_REL`.
+pub fn test_completeness_generic_only_relation<const N_REL: usize, RNG>(
```
[the remark this answers is not in the document]
Comment (volhovm) [reply msm/src/test/generic.rs:12]: No strong opinion. I think it's a bit easier to understand the code if there are 5 variants of the test and they all call the generic one, so I don't have to think about dependencies between these 5 tests. But it's subjective.

**Review by dannywillems — APPROVED** [review]

**Comment by dannywillems** [comment]
Conflicts must be solved when https://github.com/o1-labs/proof-systems/pull/2249.

**Comment by dannywillems** [comment]
Taking care of fixing conflicts now.

**Comment by dannywillems** [comment]
Conflicts solved

**Pull request merged**